src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskAssignByIdControllerTest.java

```diff
@@ -201,323 +147,8 @@ public void assigner_has_grant_type_specific_and_permission_manage_own_can_assig
-
-        initiateTask(taskVariables, caseworkerForReadCredentials.getHeaders());
-
-        common.setupCaseManagerForSpecificAccess(assigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-
-        common.cleanUpTask(taskId);
-    }
-
-    @Test
-    public void assigner_has_grant_type_specific_and_permission_execute_manage_can_assign_other_already_assigned_task() {
-        //assigner role : execute, manage
-        //assignee role : own, manage
-        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "reviewSpecificAccessRequestJudiciary");
-        taskId = taskVariables.getTaskId();
-
-        common.setupFtpaJudgeForSpecificAccess(assignerCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        common.setupCFTJudicialOrganisationalRoleAssignment(caseworkerForReadCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-
-
-        initiateTask(taskVariables, caseworkerForReadCredentials.getHeaders());
-
-        //first assign
-        common.setupCaseManagerForSpecificAccess(assigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-
-        //second assign
-        common.setupCaseManagerForSpecificAccess(secondAssigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        assignTaskAndValidate(taskVariables, getAssigneeId(secondAssigneeCredentials.getHeaders()));
-
-        common.cleanUpTask(taskId);
-    }
-
-    @Test
-    public void assigner_has_grant_type_specific_and_permissions_read_manage_cancel_should_assign_a_task() {
-        //assigner role : read, manage, cancel
-        //assignee role : own, manage
-        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "reviewSpecificAccessRequestJudiciary");
-        taskId = taskVariables.getTaskId();
-
-        common.setupHearingPanelJudgeForSpecificAccess(assignerCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        common.setupCFTJudicialOrganisationalRoleAssignment(caseworkerForReadCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-
-        initiateTask(taskVariables, caseworkerForReadCredentials.getHeaders());
-
-        common.setupCaseManagerForSpecificAccess(assigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-
-        common.cleanUpTask(taskId);
-    }
-
-    @Test
-    public void assigner_has_grant_type_specific_and_permission_read_manage_cancel_can_assign_other_already_assigned_task() {
```

Comment:
This covers reassigning a task. Do we have an IT for this? Or should we retain this?

Reply:
It doesn't do anything different in assigning an already assigned task. However, in 1568, granular permission ticket, we made a change to make assignee optional. In that scenario you can UNASSIGNED a task already assigned to someone. Need to check if this is valid for exsting permission as well with Sindhu. And then need to add two test to cover that.

Reply:
This ticket is created to fix the bug https://tools.hmcts.net/jira/browse/RWA-2004